packages/web-app-files/tests/unit/components/Upload/DetailsWidget.spec.js

```diff
@@ -0,0 +1,77 @@
+import { shallowMount, createLocalVue } from '@vue/test-utils'
+import DesignSystem from 'owncloud-design-system'
+
+import DetailsWidget from '@files/src/components/Upload/DetailsWidget.vue'
+
+const localVue = createLocalVue()
+localVue.use(DesignSystem)
+
+const selectors = {
+  oCProgress: 'oc-progress-stub',
+  itemName: '#item-name',
+  itemSize: '#item-size',
+  progressText: '.oc-progress-text'
+}
+
+const items = {
+  id: 1,
+  name: 'lorem.txt',
+  size: '100',
+  progress: 20
+}
+
+describe('DetailsWidget component', () => {
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('prop "items" is required', () => {
+    expect(DetailsWidget.props.items.required).toBe(true)
+  })
+
+  describe('when an item is provided in prop', () => {
+    jest
+      .spyOn(DetailsWidget.mixins[1].methods, 'getResourceSize')
+      .mockImplementation(size => size + ' B')
+    jest
+      .spyOn(DetailsWidget.mixins[0].filters, 'roundNumber')
+      .mockImplementation(progress => parseInt(progress.toFixed(0)))
```

Comment:
why are you mocking these functions. IMO these should work without mocking. Also, these may affect your tests if the code implementations change.

Reply:
I always mock functions not implemented in the component to reduce test failures that may occur due to errors in external functions (also isolating the component).
Maybe not mocking might be good approach. Need suggestions

Reply:
IMO we should mock function that acts as input to the component. that's how we feed data for the component. also to spy its good, but for functions other than that, mocking should be least used.

Reply:
@kiranparajuli589 I have made changes please review